addons/sourcemod/scripting/shop.sp

```diff
@@ -1198,6 +1204,338 @@ public int ItemPanel_Handler(Menu menu, MenuAction action, int param1, int param
+		
+		bool isHidden = ItemManager_GetItemHideEx(sItemId);
+		
+		SetGlobalTransTarget(client);
+		
+		int credits = GetCredits(client);
+		
+		FormatEx(sBuffer, sizeof(sBuffer), "%t\n ", "credits", credits);
+		panel.SetTitle(sBuffer, false);
+		
+		ItemType type = ItemManager_GetItemTypeEx(sItemId);
+	
+		panel.DrawItem(" ", ITEMDRAW_SPACER|ITEMDRAW_RAWLINE);
+
+		int button = 1;
+		
+		switch (type)
+		{
+			case Item_None :
+			{
+				if (!isHidden)
+				{
+					FormatEx(sBuffer, sizeof(sBuffer), "%t", "You Buy Sure");
+					panel.DrawText(sBuffer);
+
+					FormatEx(sBuffer, sizeof(sBuffer), "%t", "Yes");
+					panel.DrawItem(sBuffer);
+					iConfirmButton[client][button++] = CONFIRM_YES;
+
+					FormatEx(sBuffer, sizeof(sBuffer), "%t", "No");
+					panel.DrawItem(sBuffer);
+					iConfirmButton[client][button++] = CONFIRM_NO;
+				}
+			}
+			case Item_Finite :
+			{
+				int count = PlayerManager_GetItemCountEx(client, sItemId);
+				FormatEx(sBuffer, sizeof(sBuffer), "%t: %d", "You have", count);
+				panel.DrawText(sBuffer);
+				
+				panel.DrawItem(" ", ITEMDRAW_SPACER|ITEMDRAW_RAWLINE);
+				
+				if (!isHidden)
+				{
+					FormatEx(sBuffer, sizeof(sBuffer), "%t", "You Buy Sure");
+					panel.DrawText(sBuffer);
+
+					FormatEx(sBuffer, sizeof(sBuffer), "%t", "Yes");
+					panel.DrawItem(sBuffer);
+					iConfirmButton[client][button++] = CONFIRM_YES;
+
+					FormatEx(sBuffer, sizeof(sBuffer), "%t", "No");
+					panel.DrawItem(sBuffer);
+					iConfirmButton[client][button++] = CONFIRM_NO;
+				}
+			}
+			case Item_Togglable :
+			{
+				if (!isHidden)
+				{
+					FormatEx(sBuffer, sizeof(sBuffer), "%t", "You Buy Sure");
+					panel.DrawText(sBuffer);
+
+					FormatEx(sBuffer, sizeof(sBuffer), "%t", "Yes");
+					panel.DrawItem(sBuffer);
+					iConfirmButton[client][button++] = CONFIRM_YES;
+
+					FormatEx(sBuffer, sizeof(sBuffer), "%t", "No");
+					panel.DrawItem(sBuffer);
+					iConfirmButton[client][button++] = CONFIRM_NO;
+				}
+			}
+			case Item_BuyOnly :
+			{
+				if (!isHidden)
+				{
+					FormatEx(sBuffer, sizeof(sBuffer), "%t", "You Buy Sure");
+					panel.DrawText(sBuffer);
+
+					FormatEx(sBuffer, sizeof(sBuffer), "%t", "Yes");
+					panel.DrawItem(sBuffer);
+					iConfirmButton[client][button++] = CONFIRM_YES;
+
+					FormatEx(sBuffer, sizeof(sBuffer), "%t", "No");
+					panel.DrawItem(sBuffer);
+					iConfirmButton[client][button++] = CONFIRM_NO;
+				}
+			}
+		}
+		
+		panel.DrawItem(" ", ITEMDRAW_SPACER|ITEMDRAW_RAWLINE);
+		
+		panel.CurrentKey = g_iMaxPageItems-2;
+		iConfirmButton[client][g_iMaxPageItems-2] = CONFIRM_BACK;
+		FormatEx(sBuffer, sizeof(sBuffer), "%t", "Back");
+		panel.DrawItem(sBuffer, ITEMDRAW_CONTROL);
+		
+		panel.DrawItem(" ", ITEMDRAW_SPACER|ITEMDRAW_RAWLINE);
+		
+		panel.CurrentKey = g_iMaxPageItems;
+		iConfirmButton[client][g_iMaxPageItems] = CONFIRM_EXIT;
+		FormatEx(sBuffer, sizeof(sBuffer), "%t", "Exit");
+		panel.DrawItem(sBuffer, ITEMDRAW_CONTROL);
```

Comment:
just use Menu intead Panel. Then you don't need to store state for each players

Reply:
@R1KO yes, but if you want to control what to draw, better to use Panel, to minimize amount of selectable items.

Reply:
@R1KO I was trying to also display item info on confirm panel, is that possible to overlap with menu and panel?

Reply:
Exit and Back button is unnecessary, so I removed them.

Reply:
> @R1KO I was trying to also display item info on confirm panel, is that possible to overlap with menu and panel?

Only one menu/panel can be shown to user at once.